packages/xod-arduino/src/transpiler.js

```diff
@@ -390,11 +391,11 @@ const transformProjectWithImpls = def(
       Project.extractBoundInputsToConstNodes(R.__, path, project),
     )),
     R.chain(Project.flatten(R.__, path, impls)),
-    R.ifElse(
+    R.when(
       () => (opts.debug === true),
-      Either.of,
-      Project.updatePatch(path, Project.removeDebugNodes)
-    )
+      R.chain(Project.updatePatch(path, Project.removeDebugNodes))
+    ),
```

Comment:
O_o. When debug, remove debug nodes? I think an exact opposite of it is required. Would be simpler to just:

```javascript
proj => opts.debug ? R.chain(Project.updatePatch(path, Project.removeDebugNodes), proj) : proj;
```

Please, verify debug mode works correctly after the change.